### PR TITLE
Restructuring create methods and adding helper

### PR DESCRIPTION
**Pull request by missjantastic**
- Move all create methods to corresponding services
- Add validation helper to utils

### CODE REVIEW

File: server/services/communities-service.js [hunk 1]
```diff
@@ -1,5 +1,24 @@
...
 
+/**
+ * 
+ * @param {Object} object an object containing fields for community
+ * @param {Object.key -- Number} serverId 
+ * @param {Object.key -- Array[Ornaments]} foundOrnaments
+ * @param {Object.key -- Array[Creature]} foundCreatures
+ */
+async function createCommunity({serverId, foundOrnaments, foundCreatures}) {
+    validateRequiredFields({serverId}, 'community');
```
Comment (tangyyyyyyy) [inline server/services/communities-service.js:13]: we love error handling

File: server/services/creatures-service.js [hunk 2]
```diff
@@ -1 +1,14 @@
-// createCreature()
\ No newline at end of file
+const { createDatabaseEntry, validateRequiredFields } = require('../utils/helper-functions')
+const { Creature } = require('../schemas/creature')
+
+async function createCreature({name, pronoun, items, nature, image, isFound, foundBy}) {
+    validateRequiredFields({name, pronoun, items}, 'creature');
```
Comment (tangyyyyyyy) [inline server/services/creatures-service.js]: nature is also required

File: server/services/creatures-service.js [hunk 3]
```diff
@@ -1 +1,14 @@
...
+
+    const creature = new Creature({name, pronoun, items, nature, image, isFound, foundBy})
+
+    if(creature) {
+        await createDatabaseEntry(creature);
+    }
+}
+
+module.exports = { createCreature }
```
Comment (tangyyyyyyy) [inline server/services/creatures-service.js:26]: wonderful

File: server/utils/helper-functions.js [hunk 4]
```diff
@@ -8,11 +8,23 @@
 async function createDatabaseEntry(databaseEntry) {
     try {
         await databaseEntry.save()
-        console.log(`Added ${databaseEntry?.name} to the ${databaseEntry.collection.collectionName} collection` || `Added object to the ${databaseEntry.collection.collectionName} collection`);
+        const successMessage = databaseEntry?.name ? `Added ${databaseEntry?.name} to the ${databaseEntry.collection.collectionName} collection` : `Added object to the ${databaseEntry.collection.collectionName} collection`;
```
Comment (tangyyyyyyy) [inline server/utils/helper-functions.js:11]: powerful

**Review by tangyyyyyyy — CHANGES_REQUESTED** [review]
add nature -> get approved

**Review by tangyyyyyyy — APPROVED** [review]

**Review by tangyyyyyyy — APPROVED** [review]

File: server/server.js [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+console.log("INITIALIZING CHRISTMAS BOT");
+
+async function main() {
+    console.log("The elves are getting to work...")
+    await connect(databaseToken, {dbName: 'christmas_bot_v2'}).catch(console.error);
+    console.log("Christmas Bot Status: ONLINE");
+
+
+    //TESTING: set to true if you want to purge databases
+    let input = true
```
Comment (Awood1029) [inline server/server.js:27]: What's this for?

Comment (Awood1029) [reply server/server.js:27]: Oh oops, I see you were just moving the file and it was already there.
Still should do something about this, not sure what the intent of hard-coding this to true is

Comment (missjantastic) [reply server/server.js:27]: @tangyyyyyyy to explain his reasoning

I think it's a dev-only flag so to purge the database when the code runs. I suppose the same can be accomplished by choosing to comment out/not comment out the code

Comment (Awood1029) [reply server/server.js:27]: You can directly do
If (true) or if (false), doesn't need to be set into a variable.

But if a variable is gonna be used you should name it something other than input, maybe something like "purgeDatabaseOnStartup"

Going down train of thought, a few things to consider..

Could create a seedDatabase() function that drop purges the database on startup, but also creates fresh test data if needed that would be consistent everytime. Might not be the intent now, but at some point we'd probably want some existing test data without having to add it manually each time.

**Pull request merged**